### PR TITLE
Calendroid: initial recycler view

### LINKED ISSUE #4282: Create the CalendarRecyclerView
This RecyclerView will be responsible for drawing the calendar background.

What is in the background?
- The hour indicators labels
- The hour indicators horizontal lines
- The start and end labels of Events while in edit mode (either moving or changing durations)

This can be done by overriding the `OnDraw` method in the `RecyclerView` and drawing the lines and labels.
Don't forget to call `base.OnDraw` as well.

### PR DESCRIPTION
**Pull request by daividssilverio**
## What's this?
This is the recycler view that will be used to draw the calendar events.
Its custom drawing handles the calendar hour lines and hour labels.
It still doesn't draw the hour labels while in edit mode.

### Relationships
Closes #4282
I've added another issue to complete the hour labels drawing while in edit mode (#4398).
I rather have it like this so I can work on the custom LayoutManager first.

## Why do we want this?
To have the android calendar.

## How is it done?
It overrides the `onDraw` method of the recycler view. That's it.

### Why not another way?
This seemed simple enough.

### Side effects
None.

## Review hints
As it is, it's troublesome to see it working. You'd have to create a custom adapter with 24 dummy items, wrap the recycler view in a `NestedScrollView`, modify some screen to add it to the layout.

But it looks like this, except by the hour labels which are properly formatted now:
image

## :squid: Permissions
🦑 please

### CODE REVIEW

File: Toggl.Giskard/Views/CalendarRecyclerView.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+namespace Toggl.Giskard.Views
+{
+    [Register("toggl.giskard.views.CalendarRecyclerView")]
+    public class CalendarRecyclerView : RecyclerView
+    {
+        private const int hoursPerDay = 24;
+
+        private static readonly string twelveHoursFormat = "h tt";
+        private static readonly string twentyFourHoursFormat = "HH:mm";
```
Comment (heytherewill) [inline Toggl.Giskard/Views/CalendarRecyclerView.cs]: Aren't these constants defined elsewhere? ping @juanlaube

Comment (daividssilverio) [reply Toggl.Giskard/Views/CalendarRecyclerView.cs]: Yeah, here:
https://github.com/toggl/mobileapp/blob/545292e8741f89fe6a0d2041d2afdb31b979e4d0/Toggl.Daneel/ViewSources/CalendarCollectionViewSource.cs#L25-L32

Would like to see it moved to `Resources.resx` or have some other file in Foundation?

Comment (juanlaube) [reply Toggl.Giskard/Views/CalendarRecyclerView.cs]: We should put these in Resources.resx if possible

**Comment by juanlaube** [comment]
I didn't test it because I need to setup a lot of things, but the code looks good to me.

**Comment by heytherewill** [comment]
@daividssilverio Do you mind moving the things to another file so we can merge this? :)

File: Toggl.Giskard/Views/CalendarRecyclerView.cs (hunk 1 above)
Comment (daividssilverio) [reply Toggl.Giskard/Views/CalendarRecyclerView.cs]: Moved!

**Comment by daividssilverio** [comment]
@heytherewill @juanlaube , I've moved the strings to Resources.resx.
I didn't move the integers, I'll do it when I get there, but I don't know where to place them 🤔

**Comment by daividssilverio** [comment]
@heytherewill said internally he approved this, I guess he forgot to press the button here. I'll just merge it.

**Pull request merged**